src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    public static final int INITIAL_SIZE = 10;
+
+    private T[] array = (T[]) new Object[INITIAL_SIZE];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        checking();
+        array[size++] = value;
```

Comment:
The `checking` method name is not informative. Consider renaming it to something like `ensureCapacity` to better reflect its purpose.

Reply:
Done